src/main/kotlin/org/expo/nothanks/config/properties/DefaultGameProperties.kt

```diff
@@ -0,0 +1,15 @@
+package org.expo.nothanks.config.properties
+
+import org.springframework.boot.context.properties.ConfigurationProperties
+import org.springframework.boot.context.properties.ConstructorBinding
+
+@ConfigurationProperties("no-thanks.default-game-params")
+@ConstructorBinding
+data class DefaultGameProperties (
+    val defaultCoinsCount: Int,
+    val minCard: Int,
+    val maxCard: Int,
+    val extraCards: Int,
+    val maxPlayerNumber: Int,
+    val coinsMap: Map<Int, Int>
+)
```

Comment:
According to @Steellemm, the magic of Spring when working with the config was supposed to make our lives easier, instead you were forced to create absolutely the same dataclass as `GameParams` + a bunch of additional methods just to support this solution (for example, `GamesService.getGameParams`).

If we followed the path of default values from the config, in my opinion, it should work as follows:
the GameParams data class should pull up default values from the config, thus making calling the `GameParams()` constructor without arguments absolutely legal. 
Next, in the `GameService` constructor, we must provide default GameParams() without autowiring:
```
class GamesService(val gameParams: GameParams = GameParams()) {
```
I'm not very familiar with Spring, so I can't provide a ready-made instruction. I tried using @DefaultValue in `GameParams`, but failed. I also don't see any suitable alternative to `@ConfigurationProperties` to create a single default instance of GameParams. Perhaps @Steellemm  knows other options.

But as far as I understand, we can still get rid of the original `GameParams`, rename `DefaultGameProps` to `GameParams`, and use this everywhere instead of the original.

Reply:
please have mercy on me

Reply:
Անդրեյից ոչ մի ողորմություն

Reply:
OK, nevermind

Reply:
The reason why `DefaultGameProps` and `GameParams` exist is that there is a prospected functionality of changing the options of the game during the game. It is not yet implemented (but there is some groundwork in the code already). however the distinction between `DefaultGameProps` and `GameParams` is needed for it -
`DefaultGameProps` is an immutable singleton for the whole application, while each lobby has its own mutable `GameParams` instance.
The functionality is not implemented yet, but is collapse these classes into one, we'll block option changing in the future.